pallets/polkadex-ido/src/lib.rs

```diff
@@ -125,1451 +63,628 @@ type BalanceOf<T> =
+                token_a_price_per_token_b,
+                close_round_block,
+                actual_raise: Zero::zero(),
+            }
+        }
+
+        pub fn to_primitive(&self) -> FundingRoundWithPrimitives<T::AccountId> {
+            FundingRoundWithPrimitives {
+                token_a: StringAssetId::from(self.token_a),
+                creator: self.creator.clone(),
+                amount: self.amount.saturated_into(),
+                token_b: StringAssetId::from(self.token_b),
+                vesting_per_block: self.vesting_per_block.saturated_into(),
+                start_block: self.start_block.saturated_into(),
+                vesting_end_block: self.vesting_end_block.saturated_into(),
+                project_info_cid: self.project_info_cid.clone(),
+                min_allocation: self.min_allocation.saturated_into(),
+                max_allocation: self.max_allocation.saturated_into(),
+                token_a_price_per_token_b: self.token_a_price_per_token_b.saturated_into(),
+                close_round_block: self.close_round_block.saturated_into(),
+                actual_raise: self.actual_raise.saturated_into(),
+            }
+        }
+    }
+
+    #[pallet::pallet]
+    #[pallet::without_storage_info]
+    pub struct Pallet<T>(PhantomData<T>);
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+
+        /// Registers a funding round with the amount as the total allocation for this round and vesting period.
+        ///
+        /// # Parameters
+        /// * `cid` : IPFS cid
+        /// * `token_a`: The Project token
+        /// * `amount`: Amount for funding round
+        /// * `token_b`: Token in which funding is received
+        /// * `vesting_per_block`: Vesting per block
+        /// * `funding_period`: Number of blocks from the current block for funding/show interest in funding round
+        /// * `min_allocation`: Minimum allocation of funds investor can invest
+        /// * `max_allocation`: Maximum allocation of funds investor can invest
+        /// * `token_a_price_per_token_b`: Priceper amount for project token
+        #[pallet::weight((10_000, DispatchClass::Normal))]
+        pub fn register_round(
+            origin: OriginFor<T>,
+            cid: Vec<u8>,
+            token_a: AssetId,
+            amount: BalanceOf<T>,
+            token_b: AssetId,
+            vesting_per_block: BalanceOf<T>,
+            funding_period: T::BlockNumber,
+            min_allocation: BalanceOf<T>,
+            max_allocation: BalanceOf<T>,
+            token_a_price_per_token_b: BalanceOf<T>,
+        ) -> DispatchResult {
+            let team: T::AccountId = ensure_signed(origin)?;
+            //TODO check if funder have the token_a available and reserve them.
+            // CID len must be less than or equal to 100
+            ensure!(cid.len() <= 100, <Error<T>>::CidReachedMaxSize);
```

Comment:
Configure 100 as a constant in a constants file.

Reply:
Done ✅